rfcs/2024-05-21-ui-component-model.md

```diff
@@ -0,0 +1,248 @@
+# UI component model
+
+Authors: Gordon Brander
+
+## Overview
+
+Context: converge on a default UI component model for LLM-generated UI.
+
+In a FE context, “UI component” means a bundle of UI and behavior that together describe an island of interactivity. There are various popular component models, including React, WebComponents, Vue, etc. In keeping with common FE terminology, this rfc uses the term “component” to mean “bundle of code that describes an island of UI”. For our specific context, a UI component is probably a type of module.
+
+## Goals
+
+### Product goals
+
+- Easy for LLMs to generate
+    - Leverages patterns and/or frameworks that are widely present in the training data, or can be learned within a small context window.
+- Easy for humans to edit
+    - [Maybe it’s not even code?](https://x.com/threepointone/status/1792930000766677034) Or maybe it’s a hybrid of code and plain language?
+- Leverages familiar or established patterns for UI development
+- Conformable with existing web FE toolchains.
```

Comment:
What does that mean / what kind of tool chains?

At a broad level I agree, but I also imagine that we'll quickly create a happy path for simple components that are whatever toolchain we pick.